ArrayIndexOutOfBoundsException

Sometimes(...) I get these exceptions running in IDEA 2018.3.1, Maven 3.6.0, and on Java 11:
```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 51 out of bounds for length 49
    at java.util.ArrayList.add (ArrayList.java:486)
    at java.util.ArrayList.add (ArrayList.java:498)
    at com.coveo.AbstractFMT.formatSourceFile (AbstractFMT.java:200)
    at com.coveo.AbstractFMT.lambda$formatSourceFilesInDirectory$2 (AbstractFMT.java:142)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
    at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:177)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:195)
    at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:177)
    at java.util.ArrayList$ArrayListSpliterator.forEachRemaining (ArrayList.java:1654)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:484)
    at java.util.stream.ForEachOps$ForEachTask.compute (ForEachOps.java:290)
    at java.util.concurrent.CountedCompleter.exec (CountedCompleter.java:746)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:290)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec (ForkJoinPool.java:1020)
    at java.util.concurrent.ForkJoinPool.scan (ForkJoinPool.java:1656)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1594)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:177)
```

We were using Java 11, Maven 3.6.0, and fmt plugin version 2.6.0

Faced similar issue in our Gitlab CI that randomly appears:

[INFO] org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.coveo:fmt-maven-plugin:2.6.0:format (default) on project transfer-receipt-orchestrator-rms9-app: Execution default of goal com.coveo:fmt-maven-plugin:2.6.0:format failed.
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
[INFO]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
[INFO]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
[INFO]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
[INFO]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
[INFO]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
[INFO]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
[INFO]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
[INFO]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
[INFO]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
[INFO]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
[INFO]     at java.lang.reflect.Method.invoke (Method.java:566)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[INFO] Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default of goal com.coveo:fmt-maven-plugin:2.6.0:format failed.
[INFO]     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:148)
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
[INFO]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
[INFO]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
[INFO]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
[INFO]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
[INFO]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
[INFO]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
[INFO]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
[INFO]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
[INFO]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
[INFO]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
[INFO]     at java.lang.reflect.Method.invoke (Method.java:566)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[INFO] Caused by: java.lang.ArrayIndexOutOfBoundsException
[INFO]     at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
[INFO]     at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:62)
[INFO]     at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
[INFO]     at java.lang.reflect.Constructor.newInstance (Constructor.java:490)
[INFO]     at java.util.concurrent.ForkJoinTask.getThrowableException (ForkJoinTask.java:603)
[INFO]     at java.util.concurrent.ForkJoinTask.reportException (ForkJoinTask.java:678)
[INFO]     at java.util.concurrent.ForkJoinTask.invoke (ForkJoinTask.java:737)
[INFO]     at java.util.stream.ForEachOps$ForEachOp.evaluateParallel (ForEachOps.java:159)
[INFO]     at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel (ForEachOps.java:173)
[INFO]     at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:233)
[INFO]     at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:497)
[INFO]     at com.coveo.AbstractFMT.formatSourceFilesInDirectory (AbstractFMT.java:142)
[INFO]     at com.coveo.AbstractFMT.execute (AbstractFMT.java:101)
[INFO]     at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
[INFO]     at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
[INFO]     at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
[INFO]     at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
[INFO]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
[INFO]     at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
[INFO]     at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
[INFO]     at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
[INFO]     at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
[INFO]     at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
[INFO]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
[INFO]     at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
[INFO]     at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
[INFO]     at java.lang.reflect.Method.invoke (Method.java:566)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
[INFO]     at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
[INFO] Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 0
[INFO]     at java.util.ArrayList.add (ArrayList.java:486)
[INFO]     at java.util.ArrayList.add (ArrayList.java:498)
[INFO]     at com.coveo.AbstractFMT.formatSourceFile (AbstractFMT.java:200)
[INFO]     at com.coveo.AbstractFMT.lambda$formatSourceFilesInDirectory$2 (AbstractFMT.java:142)
[INFO]     at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
[INFO]     at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:177)
[INFO]     at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:195)
[INFO]     at java.util.stream.ReferencePipeline$2$1.accept (ReferencePipeline.java:177)
[INFO]     at java.util.ArrayList$ArrayListSpliterator.forEachRemaining (ArrayList.java:1654)
[INFO]     at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:484)
[INFO]     at java.util.stream.ForEachOps$ForEachTask.compute (ForEachOps.java:290)
[INFO]     at java.util.concurrent.CountedCompleter.exec (CountedCompleter.java:746)
[INFO]     at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:290)
[INFO]     at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec (ForkJoinPool.java:1020)
[INFO]     at java.util.concurrent.ForkJoinPool.scan (ForkJoinPool.java:1656)
[INFO]     at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1594)
[INFO]     at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:177)
[INFO] [ERROR] 
[INFO] [ERROR]

--- a/src/main/java/com/coveo/AbstractFMT.java
+++ b/src/main/java/com/coveo/AbstractFMT.java
@@ -13,13 +13,13 @@ import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
-import org.apache.maven.plugin.logging.Log;
 import org.apache.maven.plugins.annotations.Parameter;
 
 public abstract class AbstractFMT extends AbstractMojo {
@@ -62,7 +62,7 @@ public abstract class AbstractFMT extends AbstractMojo {
   @Parameter(defaultValue = "google", property = "style")
   private String style;
 
-  private List<String> filesProcessed = new ArrayList<String>();
+  private List<String> filesProcessed = new CopyOnWriteArrayList<>();
   private int nonComplyingFiles;
 
   /**

